src/app/components/sandbox/Preview/index.js

```diff
@@ -56,20 +57,27 @@ type State = {
 };
 
 export default class Preview extends React.PureComponent {
+  initialPath: string
+  
   constructor(props: Props) {
     super(props);
 
     this.state = {
       frameInitialized: false,
       history: [],
       historyPosition: 0,
-      urlInAddressBar: '',
+      urlInAddressBar: props.initialPath || '',
       url: null,
     };
 
     if (!props.noDelay) {
       this.executeCode = debounce(this.executeCode, 800);
     }
+
+    // we need a value that doesn't change when receiving `initialPath`
+    // from the query params, or the iframe will continue to be re-rendered
+    // when the user navigates the iframe app, which shows the loading screen
+    this.initialPath = this.state.urlInAddressBar
```

Comment:
I meant navigating when the app uses history.push like my sandbox. It was refreshing because the URL kept changing. It kept changing because no longer was passed just the base URL. 

So I fixed it by making sure it got the URL, but only once on mount. That way it can receive initialPath from query params. It doesn't need to think about it anymore after the first render. Put back urlInAddressBar and try clicking links in my codesandbox to see.